### PR TITLE
Spl and Marinade-lp clients & tests + main client introduction(wip)

### PR DESCRIPTION
**Pull request by 0xcrust**

### CODE REVIEW

File: packages/sdks/spl/src/getStakePool.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+/**
+ * TODO: FIX DEPENDENCY CONFLICTS TO PREVENT HAVING TO DO THIS.
```
Comment (dankelleher) [inline packages/sdks/spl/src/getStakePool.ts:2]: If we upgrade spl-token across the board will it solve this? I think marinade-sdk uses an old version but was recently updated to use @solana/spl-token-3.x

File: packages/sdks/spl/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+  canDepositSol,
+} from "../../sunrise-stake-client/src/beamInterface";
+import BN from "bn.js";
+import { SunriseClient } from "../../sunrise/src";
+
+import {
+  MarinadeUtils,
+  Provider,
+  type Wallet,
+} from "@sunrisestake/marinade-ts-sdk";
```
Comment (dankelleher) [inline packages/sdks/spl/src/index.ts:39]: Can we avoid using marinade in the spl beam?

File: packages/sdks/spl/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+    programId?: PublicKey
+  ): Promise<SplClient> {
+    let PID = programId ?? SPL_BEAM_PROGRAM_ID;
+    const state = Utils.deriveStateAddress(PID, sunriseState)[0];
+
+    const client = await this.get(state, provider, splStakePool, PID, true);
+    const vaultAuthority = client.vaultAuthority[0];
+    const vaultAuthorityBump = client.vaultAuthority[1];
+
+    let stakePool = client.stakePool;
```
Comment (dankelleher) [inline packages/sdks/spl/src/index.ts:142]: prefer `const` for immutable variables

File: programs/sunrise_beam/tests/helpers/context.rs [hunk 4]
```diff
@@ -0,0 +1,179 @@
+use super::{instructions::*, Result, SunriseContextError};
```
Comment (dankelleher) [inline programs/sunrise_beam/tests/helpers/context.rs:1]: really nice helper mod.

**Review by dankelleher — APPROVED** [review]
Fantastic PR @0xcrust , great work!

File: packages/sdks/marinade-sp/src/index.ts [hunk 5]
```diff
@@ -44,18 +52,26 @@ export class MarinadeClient implements BeamInterface {
   /** The address of the authority of this beam's token vaults*/
   vaultAuthority: [PublicKey, number];
 
+  /** Fields that depend on the stake-pool state. */
```
[the remark this answers is not in the document]
Comment (0xcrust) [reply packages/sdks/marinade-sp/src/index.ts:55]: Thanks!

File: packages/sdks/spl/src/getStakePool.ts (hunk 1 above)
Comment (0xcrust) [reply packages/sdks/spl/src/getStakePool.ts:2]: I tried fixing it but ran into multiple issues and so just left it like this to save time. It's probably fixable though

**Pull request merged**